perun-notification/src/main/java/cz/metacentrum/perun/notif/managers/PerunNotifEmailManagerImpl.java

```diff
@@ -50,278 +30,38 @@ public class PerunNotifEmailManagerImpl implements PerunNotifEmailManager {
-					}
-				}
-			}
-			logger.error("Sending of the email failed.", ex);
-		} catch (Exception ex) {
-			throw new EmailPreparationException(ex);
-		}
-	}
-
-	/**
-	 * Actually send the given array of MimeMessages via JavaMail.
-	 *
-	 * @param mimeMessages MimeMessage objects to send
-	 * @throws EmailAuthenticationException in case of authentication
-	 * failure
-	 * @throws EmailSendException in case of failure when sending a message
-	 */
-	protected void doSend(MimeMessage[] mimeMessages, List<String> contents) {
-		Map<Object, Exception> failedMessages = new LinkedHashMap<Object, Exception>();
-
-		Transport transport;
 		try {
-			transport = getTransport(getSession());
-			transport.connect(smtpHost, port, username, password);
-		} catch (AuthenticationFailedException ex) {
-			throw new EmailAuthenticationException(ex);
-		} catch (MessagingException ex) {
-			// Effectively, all messages failed...
-			for (int i = 0; i < mimeMessages.length; i++) {
-				failedMessages.put(mimeMessages[i], ex);
-			}
-			throw new EmailSendException("Mail server connection failed", ex, failedMessages);
-		}
-
-		try {
-			for (int i = 0; i < mimeMessages.length; i++) {
-				MimeMessage mimeMessage = mimeMessages[i];
-				String content = contents.get(i);
+			JavaMailSender mailSender = BeansUtils.getDefaultMailSender();
+			for (PerunNotifEmailMessageToSendDto dto : list) {
+				SimpleMailMessage message = new SimpleMailMessage();
+				message.setFrom(dto.getSender());
+				message.setTo(dto.getReceiver());
+				message.setSubject(dto.getSubject());
+				message.setText(dto.getMessage());
 				try {
-					if (mimeMessage.getSentDate() == null) {
-						mimeMessage.setSentDate(new Date());
-					}
-					mimeMessage.saveChanges();
-					transport.sendMessage(mimeMessage, mimeMessage.getAllRecipients());
-
-
-					sendMessagesLogger.info("Email sent in {} with receivers: "
-						+  Arrays.toString(mimeMessage.getRecipients(Message.RecipientType.TO))
-						+ " senders: " + Arrays.toString(mimeMessage.getFrom())
-						+ " subject: " + mimeMessage.getSubject()
-						+ " content: "
-						+ content,
-						new Date());
-				} catch (MessagingException ex) {
-					try {
-						logger.error("Error during send of email for: {}", mimeMessage.getRecipients(Message.RecipientType.TO), ex);
-					} catch (Exception ex2) {
-						logger.error("Cannot send email.", ex);
-					}
-					failedMessages.put(mimeMessage, ex);
-				}
-			}
-		} finally {
-			try {
-				transport.close();
-			} catch (MessagingException ex) {
-				if (!failedMessages.isEmpty()) {
-					throw new EmailSendException("Failed to close server connection after message failures", ex, failedMessages);
-				} else {
-					throw new EmailSendException("Failed to close server connection after message sending", ex);
+					//send message over SMTP
+					mailSender.send(message);
+					// log successful sending
+					sendMessagesLogger.info("Email sent date: {}, to: {}, from: {}, subject: {}, text: {}",
```

Comment:
Since we don't set `sentDate` anymore we shouldn't try to log it, as it is always null.